sys/shell/shell.c

```diff
@@ -168,7 +171,13 @@ static void handle_input_line(const shell_command_t *command_list, char *line)
     int argc = 0;
     char *readpos = line;
     char *writepos = readpos;
-    enum PARSE_STATE pstate = PARSE_SPACE;
+
+    enum parse_state pstate = PARSE_SPACE;
```

Comment:
Using enum types seems like an overhear for something that can be covered with `uint8_t`.

Reply:
It could also be handled by more simple data types, but then we can't collect the states like it's done with the enum. I like it to have them stick together at the 'same spot'. Is there any drawback when using enum here?

Reply:
Depending on the arch, enum are 4byte long I think. What do you mean by collecting the states?

Reply:
By collecting I mean that values with the same purpose, the states, are bounded in a single data structure and not spread over mutliple single definitions. For me an enum improves readability because one can see that the interior values serve the same purpose. So if a single enum entry is 4 byte it wastes 3 bytes compared to an `uint8_t` which sums up to 7*3 = 21 bytes difference. Is that worth changing it? If it's considered valuable then I wont block this change.

Reply:
Oh, or did you really just mean the variable `pstate`?

Reply:
Yes I just meant the type of the `pstate` variable.

Reply:
Changed it to `uint8_t` :)